source/gameengine/GameLogic/Joystick/SCA_Joystick.cpp

```diff
@@ -77,92 +75,74 @@ SCA_Joystick *SCA_Joystick::m_instance[JOYINDEX_MAX];
 int SCA_Joystick::m_joynum = 0;
 int SCA_Joystick::m_refCount = 0;
 
-SCA_Joystick *SCA_Joystick::GetInstance( short int joyindex )
+void SCA_Joystick::Init()
 {
-#ifndef WITH_SDL
-	return NULL;
-#else  /* WITH_SDL */
-	if (!SDL_CHECK(SDL_InitSubSystem)) {
-		return NULL;
-	}
-	if (joyindex < 0 || joyindex >= JOYINDEX_MAX) {
-		JOYSTICK_ECHO("Error-invalid joystick index: " << joyindex);
-		return NULL;
-	}
+#ifdef WITH_SDL
 
-	if (m_refCount == 0) 
-	{
-		int i;
+	if (!(SDL_CHECK(SDL_InitSubSystem)) ||
+	    !(SDL_CHECK(SDL_GameControllerAddMapping)) ||
+	    !(SDL_CHECK(SDL_NumJoysticks))) {
+		return;
+	}
 
-		/* The video subsystem is required for joystick input to work. However,
-		 * when GHOST is running under SDL, video is initialized elsewhere. We
-		 * also need to set the videodriver to dummy, and do it here to avoid
-		 * interfering with addons that may use SDL too.
-		 *
-		 * We also init SDL once only. */
-#  ifdef WITH_GHOST_SDL
```

Comment:
In all net example about SDL2 joystick that i can find it was not necessity to enable SDL video subsystem.